joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -285,14 +287,10 @@ controller_interface::return_type JointTrajectoryController::update(
         }
         if (has_effort_command_interface_)
         {
-          if (use_closed_loop_pid_adapter_)
+          if (use_closed_loop_pid_adapter_)  // this should be always set to true when using effort
```

Comment:
I think a table of which combinations of these things are valid would be great. It's hard to follow with comments scattered around the code. Or I guess you could use a std::map of which combinations are valid.

- has_position_command_interface_
- has_velocity_command_interface_
- has_effort_command_interface_
- use_closed_loop_pid_adapter_
          
 Maybe you should add these checks:
 
```
 if (has_effort_command_interface_ && !use_closed_loop_pid_adapter_)
   // error
if (has_effort_command_interface_ && !has_position_command_interface)
  // error
if (has_effort_command_interface_ && has_velocity_interface_)
  // error
```